packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/PreviousMarks/PreviousMarksConnector.spec.js

```diff
@@ -0,0 +1,179 @@
+    }
+  },
+  subjectViewer: {
+    frame: 1
+  }, 
+  workflowSteps: {
+    interactionTask: {}
+  }
+}
+
+const drawingAnnotation = {
+  id: cuid(),
+  task: 'T0',
+  taskType: 'drawing'
+}
+
+const transcriptionAnnotation = {
+  id: cuid(),
+  task: 'T0',
+  taskType: 'transcription'
+}
+
+const mockStoresWithDrawingAnnotations = {
+  classifications: {
+    active: {
+      interactionTaskAnnotations: [drawingAnnotation]
+    }
+  },
+  subjectViewer: {
+    frame: 1
+  },
+  workflowSteps: {
+    interactionTask: {
+      shownMarks: 'USER',
+      type: 'drawing'
+    }
+  }
+}
+
+const mockStoresWithTranscriptionAnnotations = {
+  classifications: {
+    active: {
+      interactionTaskAnnotations: [transcriptionAnnotation]
+    }
+  },
+  subjectViewer: {
+    frame: 1
+  },
+  workflowSteps: {
+    interactionTask: {
+      shownMarks: 'USER',
+      type: 'transcription'
+    }
+  }
+}
+
+describe('Component > PreviousMarksConnector', function () {
+  let mockUseContext
+  afterEach(function () {
+    mockUseContext.restore()
+  })
+
+  it('should render without crashing', function () {
+    mockUseContext = sinon.stub(React, 'useContext').callsFake(() => {
+      return {
+        classifierStore: mockStores
+      }
+    })
+    const wrapper = shallow(<PreviousMarksConnector />)
+    expect(wrapper).to.be.ok()
+  })
+
+  it('should pass along the frame index from the subject viewer store', function () {
+    mockUseContext = sinon.stub(React, 'useContext').callsFake(() => {
+      return {
+        classifierStore: mockStores
+      }
+    })
+    const wrapper = shallow(<PreviousMarksConnector />)
+    expect(wrapper.find(PreviousMarks).props().frame).to.equal(1)
+  })
+
+  it('should pass along the scale from props', function () {
+    mockUseContext = sinon.stub(React, 'useContext').callsFake(() => {
+      return {
+        classifierStore: mockStores
+      }
+    })
+    const wrapper = shallow(<PreviousMarksConnector scale={2} />)
+    expect(wrapper.find(PreviousMarks).props().scale).to.equal(2)
+  })
+
+  describe('when there is not an interactive task', function () {
+    it('should pass along an undefined shownMarks', function () {
+      mockUseContext = sinon.stub(React, 'useContext').callsFake(() => {
+        return {
+          classifierStore: mockStores
+        }
+      })
+      const wrapper = shallow(<PreviousMarksConnector />)
+      expect(wrapper.find(PreviousMarks).props().shownMarks).to.be.undefined()
+    })
+  })
+
+  describe('when there is an interactive task', function () {
+    it('should pass along the type of shownMarks from the drawing task', function () {
+      mockUseContext = sinon.stub(React, 'useContext').callsFake(() => {
+        return {
+          classifierStore: mockStoresWithDrawingAnnotations
+        }
+      })
+      const wrapper = shallow(<PreviousMarksConnector />)
+      expect(wrapper.find(PreviousMarks).props().shownMarks).to.equal('USER')
```

Comment:
Not for drawing tasks, but it is for transcription tasks. The `PreviousMarks` component, it is valid and needed for transcription tasks. It's impossible for the value to be set as `USER` during a drawing task since the correct binary toggle will be rendered for the control.